File: src/lib/types/expense.ts
export type CreateExpenseInput = {
    amount: number
    category_id: number
    memo?: string
    spent_at: string // yyyy-mm-dd
}

export type Expense = {
    id: number;
    amount: number;
    memo: string | null;
    spent_at: string; // YYYY-MM-DD
    category: {
        id: number;
        name: string;
    }
};

export type GetExpensesResponse = {
    expenses: Expense[];
};

File: backend/internal/domain/expense.go
package domain

type Expense struct {
	ID       int      `json:"id"`
	Amount   int      `json:"amount"`
	Memo     string   `json:"memo"`
	SpentAt  string   `json:"spent_at"`
	Status   string   `json:"status"`
	Category Category `json:"category"`
}
